File: app/src/main/java/com/example/elsafa/HomeActivity.java
package com.example.elsafa;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.example.elsafa.Search.SearchResults;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.firebase.Timestamp;
import com.google.firebase.firestore.Blob;
import com.google.firebase.firestore.FirebaseFirestore;
import com.mancj.materialsearchbar.MaterialSearchBar;
import com.mancj.materialsearchbar.SimpleOnSearchActionListener;
import com.squareup.picasso.Picasso;
import com.squareup.picasso.Target;

import java.io.ByteArrayOutputStream;

import Model.Offer;

public class HomeActivity extends AppCompatActivity {

    private MaterialSearchBar searchBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        searchBar = findViewById(R.id.search_view);

        BottomNavigationView navView = findViewById(R.id.nav_view);

        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.navigation_shop, R.id.navigation_categories, R.id.navigation_account, R.id.navigation_cart, R.id.navigation_wishlist)
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        NavigationUI.setupWithNavController(navView, navController);

        serSearchListeners();

    }

    private void serSearchListeners() {
        searchBar.setOnSearchActionListener(new SimpleOnSearchActionListener() {
            @Override
            public void onSearchConfirmed(CharSequence text) {
                super.onSearchConfirmed(text);
                Intent intent = new Intent(HomeActivity.this, SearchResults.class);
                intent.putExtra("text", String.valueOf(text));
                startActivity(intent);
            }
        });
    }

    private void setOffer(int i) {
        Uri[] imagesURL = new Uri[]{
                Uri.parse("https://image.shutterstock.com/image-vector/special-offer-banner-vector-format-600w-586903514.jpg")
                , Uri.parse("https://addconn.com/images/limited.png")};


        String[] productsIds = new String[]{
                "xZvcdItmvKWqilQ9ikbm",
                "Lnslv4jK5o7GFrITi5vm"
        };

        Picasso.get().load(imagesURL[i]).into(new Target() {
            @Override
            public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
                Timestamp time = Timestamp.now();

                Offer offer = new Offer();
                offer.setProductId(productsIds[i]);
                offer.setDiscountPercentage(33);
                offer.setStartTime(time);
                offer.setEndTime(new Timestamp(time.getSeconds() + 86400, time.getNanoseconds()));
                offer.setCategoryName("Meals");

                ByteArrayOutputStream stream = new ByteArrayOutputStream();
                bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
                offer.setOfferPic(Blob.fromBytes(stream.toByteArray()));

                FirebaseFirestore fStore = FirebaseFirestore.getInstance();
                fStore.collection("Offers").document().set(offer);
            }

            @Override
            public void onBitmapFailed(Exception e, Drawable errorDrawable) {

            }

            @Override
            public void onPrepareLoad(Drawable placeHolderDrawable) {

            }
        });
    }

}

File: app/src/main/java/com/whiteside/cafe/HomeActivity.kt
package com.whiteside.cafe

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.ui.*
import com.mancj.materialsearchbar.SimpleOnSearchActionListener
import com.whiteside.cafe.common.BaseActivity
import com.whiteside.cafe.databinding.ActivityHomeBinding
import com.whiteside.cafe.ui.search.SearchResults
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class HomeActivity : BaseActivity() {
    private lateinit var navController: NavController

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var bind: ActivityHomeBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bind = ActivityHomeBinding.inflate(layoutInflater)
        setContentView(bind.root)

        setSupportActionBar(bind.toolBar)
        setupSearchListener()

        navController = Navigation.findNavController(this, R.id.nav_host_fragment)
        appBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.navigation_shop,
                R.id.navigation_categories,
                R.id.navigation_account,
                R.id.navigation_cart,
                R.id.navigation_wishlist,
                R.id.navigation_unsigned_account,
                R.id.navigation_search
            )
        )

        if (resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE)
            setUpWithDrawer()
        else
            setUpWithToolBar()

        setUpWithBottomNav()
        setUpWithSideNav()
    }

    private fun setUpWithDrawer() {
//        val appBarConfiguration = AppBarConfiguration(navController.graph, drawer_layout)
        setupActionBarWithNavController(navController, appBarConfiguration)
    }

    private fun setUpWithBottomNav() {
        bind.bottomNav?.let {
            NavigationUI.setupWithNavController(it, navController)
        }
    }

    private fun setUpWithSideNav() {
        bind.navView?.let {
            NavigationUI.setupWithNavController(it, navController)
        }
    }

    private fun setUpWithToolBar() {
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
//        menuInflater.inflate(R.menu.tool_bar_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return item.onNavDestinationSelected(navController) || super.onOptionsItemSelected(item)
    }

    private fun setupSearchListener() {
        bind.searchView!!.setOnSearchActionListener(object : SimpleOnSearchActionListener() {
            override fun onSearchConfirmed(text: CharSequence) {
                super.onSearchConfirmed(text)
                val intent = Intent(this@HomeActivity, SearchResults::class.java)
                intent.putExtra("text", text.toString())
                startActivity(intent)
            }
        })
    }
}